### PR TITLE
W-11802236 Update OAuth Token Refresh Expression

### PR DESCRIPTION
**Pull request by valkyrie69**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: http/1.7/modules/ROOT/pages/http-authentication.adoc [hunk 1]
```diff
@@ -614,9 +614,7 @@ You can configure when Mule performs one of these requests to obtain a new acces
 . Set *Authentication* to *Authorization code grant type*.
 . Set *Refresh Token When* to *Expression*.
 . In the expression box, add the following DataWeave expression:
-`#[payload.response.status == 'unauthorized']`
-
-image::http-oauth-refreshtoken.png[Configure Response Token When field for OAuth Authentication]
+`#[attributes.statusCode == '401']`
```
Comment (eze210) [inline http/1.7/modules/ROOT/pages/http-authentication.adoc]: Without the quotes:
```suggestion
`#[attributes.statusCode == 401]`
```

File: http/1.7/modules/ROOT/pages/http-authentication.adoc [hunk 2]
```diff
@@ -633,12 +631,12 @@ In the *Configuration XML* editor, the `refreshTokenWhen` configuration looks li
         clientId="your_client_id"
         clientSecret="your_client_secret"
         tokenUrl="http://www.box.com/api/oauth2/token"
-        refreshTokenWhen="#[payload.response.status == 'unauthorized']" />
+        refreshTokenWhen="#[attributes.statusCode == '401']" />
```
Comment (eze210) [inline http/1.7/modules/ROOT/pages/http-authentication.adoc]:
```suggestion
        refreshTokenWhen="#[attributes.statusCode == 401]" />
```

File: http/1.7/modules/ROOT/pages/http-authentication.adoc [hunk 3]
```diff
@@ -807,7 +805,7 @@ The same behavior of extracting parameters from the token URL that applies to OA
 . Set *Authentication* to *Client credentials grant type*.
 . Set *Refresh Token When* to *Expression*.
 . In the expression box, add the following DataWeave expression:
-`#[payload.response.status == 'unauthorized']`
+`#[attributes.statusCode == '401']`
```
Comment (eze210) [inline http/1.7/modules/ROOT/pages/http-authentication.adoc]:
```suggestion
`#[attributes.statusCode == 401]`
```

File: http/1.7/modules/ROOT/pages/http-authentication.adoc [hunk 4]
```diff
@@ -825,7 +823,7 @@ In the *Configuration XML* editor, the `refreshTokenWhen` configuration looks li
           clientSecret="CLIENT_SECRET"
           tokenUrl="http://some.api.com/api/1.0/oauth/token"
           scopes="access_user_details, read_user_files"
-        refreshTokenWhen="#[payload.response.status == 'unauthorized']" />
+        refreshTokenWhen="#[attributes.statusCode == '401']" />
```
Comment (eze210) [inline http/1.7/modules/ROOT/pages/http-authentication.adoc]:
```suggestion
        refreshTokenWhen="#[attributes.statusCode == 401]" />
```

**Review by eze210 — DISMISSED** [review]
Don't forget to remove the quotes from the screenshot.

**Review by dhirsch1 — DISMISSED** [review]

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**